_posts/2024-08-25-mot-tracking-metrics.md

```diff
@@ -0,0 +1,172 @@
+<div class="post-center-image" style="max-width: 800px; margin: 0 auto;">
+    {% picture pimage /assets/images/fullsize/posts/2024-08-25-mot-tracking-metrics/idf1.png --alt IDF1 metric diagram %}
+</div>
+
+{:refdef: class="image-caption"}
+*Figure 3. IDF1 metric diagram*
+{: refdef}
+
+The main problem I see with IDF1 is finding the best one-to-one matching between predicted and ground truth trajectories for the entire sequence. However, this approach can sometimes oversimplify complex tracking scenarios:
+
+Imagine a corner kick in football. A tracker might correctly follow Player A running into the box, lose them in a cluster, and then mistakenly pick up Player B after the ball is cleared. IDF1 might treat this as one partially correct track for either Player A or B, ignoring that it's correct for different players at different times.
+
+This simplification can misrepresent a tracker's performance in complex situations like crowded football plays, where player interactions and occlusions are frequent.
+
+Key advantages of IDF1:
+
+1. It's more sensitive to **long-term tracking consistency**.
+2. It balances **precision and recall** of identity predictions.
+3. It's less affected by the number of objects in the scene than MOTA.
+
+However, IDF1 also has limitations:
+
+1. **IDF1 can decrease when improving detection**. Just avoiding FP can result in a better metric (A vs C)
+2. **IoU is fixed** so more or less detection accuracy is not reflected on the metric
+
+More limitations are presented in the HOTA paper. I recommend you to have a read because it is very well explained and intuitive.
+
+## HOTA (Higher Order Tracking Accuracy)
+
+HOTA is a more recent metric designed to address the limitations of both MOTA and IDF1. It aims to provide a balanced assessment of detection and association performance. HOTA can be broken down into DetA (Detection Accuracy) and AssA (Association Accuracy), allowing separate analyses of these aspects.
+
+The core HOTA formula is:
+
+$$
+ \text{HOTA}_{\alpha} = \sqrt{\text{DetA}_{\alpha} \cdot \text{AssA}_{\alpha}} 
+$$
+
+In this formula, the alpha term represents the different Intersection over Union (IoU) thresholds used to compute the metric. A True Positive (TP) is only considered when the match IoU score is above the given alpha threshold. The metric uses 19 different alpha values, ranging from 0.05 to 0.95 in increments of 0.05.
+
+HOTA uses global alignment (high-order association) between predicted and ground truth detections, similar to IDF1, but also incorporates localization accuracy. This means that HOTA evaluates both the ability to detect objects accurately and to maintain correct associations over time.
+
+The HOTA algorithm can be summarized in the following steps:
+
+```yaml
+For each frame
+	For each alpha value
+		Perform matching between gt and preds (Hungarian algorithm)
+		Obtain TP, FP and FN from previous matching
+		Compute association accuracy across the entire video for each TP.
+```
```

Comment:
I'm a bit confused about the metric. Say you have A alpha values, N frames, D detections per frame. You would compute A*N*D metrics? What's the final metric then?